source/material_model/rheology/friction_options.cc

```diff
@@ -0,0 +1,495 @@
+                                     (theta_old - critical_slip_distance / ( cellsize * current_edot_ii))
+                                     * exp( - ((current_edot_ii * cellsize) * this->get_timestep()) / critical_slip_distance);
+        return current_theta;
+      }
+
+
+      template <int dim>
+      void
+      FrictionOptions<dim>::
+      compute_theta_reaction_terms(const int q,
+                                   const MaterialModel::MaterialModelInputs<dim> &in,
+                                   const std::vector<double> &volume_fractions,
+                                   const double min_strain_rate,
+                                   const double ref_strain_rate,
+                                   const bool use_elasticity,
+                                   const bool use_reference_strainrate,
+                                   const std::vector<double> &elastic_shear_moduli,
+                                   const double dte,
+                                   MaterialModel::MaterialModelOutputs<dim> &out) const
+      {
+        //cellsize is needed for theta and the friction angle
+        double cellsize = 1;
+        if (in.current_cell.state() == IteratorState::valid)
+          {
+            cellsize = in.current_cell->extent_in_direction(0);
+          }
+
+        if (this->simulator_is_past_initialization() && this->get_timestep_number() > 0 && in.requests_property(MaterialProperties::reaction_terms) && in.current_cell.state() == IteratorState::valid)
+          {
+            for (unsigned int j=0; j < volume_fractions.size(); ++j)
+              {
+                // q war n_evaluation_points
+                const double current_edot_ii =
+                  MaterialUtilities::compute_current_edot_ii (in.composition[q], ref_strain_rate,
+                                                              min_strain_rate, in.strain_rate[q], elastic_shear_moduli[j], use_elasticity,
+                                                              use_reference_strainrate, dte);
+
+                const unsigned int theta_position_tmp = this->introspection().compositional_index_for_name("theta");
+                const double theta_old = in.composition[q][theta_position_tmp];
+                const double current_theta = compute_theta(theta_old, current_edot_ii, cellsize);
+                const double theta_increment = current_theta - theta_old;
+
+                out.reaction_terms[q][theta_position_tmp] = theta_increment;
+              }
+          }
+      }
+
+      template <int dim>
+      std::pair<double,double>
+      FrictionOptions<dim>::calculate_depth_dependent_a_and_b(const Point<dim> &position, const int j) const
+      {
+        if ( a_and_b_source == Function )
+          {
+            Utilities::NaturalCoordinate<dim> point =
+              this->get_geometry_model().cartesian_to_other_coordinates(position, coordinate_system);
```

Comment:
What coordinate system is set here? From the comments and function name, it seems like you only want depth as a function variable.